src/allotropy/parsers/nexcelom_matrix/nexcelom_matrix_reader.py

```diff
@@ -0,0 +1,27 @@
+from __future__ import annotations
+
+import pandas as pd
+
+from allotropy.allotrope.pandas_util import read_csv, read_excel
+from allotropy.exceptions import AllotropeConversionError
+from allotropy.types import IOType
+
+MILLION_CONVERSION = 1000000
+MILLION_SCALE_COLS = ["Total Cells/mL", "Live Cells/mL", "Dead Cells/mL"]
+
+
+class NexcelomMatrixReader:
+    def __init__(self, contents: IOType) -> None:
+        self.contents = contents
+        self.data = self._read_data()
+
+    def _read_data(self) -> pd.DataFrame:
+
+        try:
```

Comment:
A common pattern is to pass the whole `NamedFileContents` to the `Reader` class, then you can check the extension to determine how to read the data!

See `roche_cedex_hires_reader` for an example!

Reply:
Done in my most recent commit!